src/components/Reports/FourteenMonthReports/useFourteenMonthReport.ts

```diff
@@ -0,0 +1,127 @@
+    error,
+  } = useGetFourteenMonthReportQuery({
+    variables: {
+      accountListId,
+      // Backend sends one extra month, e.g. 13m returns 14 months
+      range: '13m',
+      designationAccountId:
+        designationAccounts && designationAccounts.length > 0
+          ? designationAccounts
+          : null,
+    },
+  });
+
+  const data =
+    currencyType === FourteenMonthReportCurrencyType.Salary
+      ? gqlData?.reportsSalaryCurrencyDonations
+      : gqlData?.reportsDonorCurrencyDonations;
+
+  const fourteenMonthReport = useMemo(() => {
+    if (!data) {
+      return null;
+    }
+
+    const isSalaryType =
+      currencyType === FourteenMonthReportCurrencyType.Salary;
+    const currencyGroups: CurrencyGroups = data.currencyGroups;
+
+    return {
+      currencyType,
+      salaryCurrency: data.defaultCurrency ?? 'USD',
+      currencyGroups: Object.entries(currencyGroups ?? {}).map(
+        ([currency, currencyGroup]: [string, CurrencyGroups[string]]) => ({
+          currency,
+          totals: {
+            year: Number(
+              isSalaryType
+                ? currencyGroup.totals.year_converted
+                : currencyGroup.totals.year,
+            ),
+            months: currencyGroup.totals.months.map((total, index) => ({
+              month: data.months[index],
+              total: Number(total),
+            })),
+            average: currencyGroup.donation_infos.reduce(
+              (averageTotal, contactDonationInfo) =>
+                averageTotal + Number(contactDonationInfo.average),
+              0,
+            ),
+            minimum: currencyGroup.donation_infos.reduce(
+              (minimumTotal, contactDonationInfo) =>
+                minimumTotal + Number(contactDonationInfo.minimum),
+              0,
+            ),
+          },
+          contacts: currencyGroup.donation_infos
+            .map((contactDonationInfo) => {
+              const contact = data.donorInfos.find(
+                (donor) => donor.contactId === contactDonationInfo.contact_id,
+              );
+              return {
+                id: contactDonationInfo.contact_id,
+                name: contact?.contactName ?? '',
+                total: Number(contactDonationInfo.total),
+                completeMonthsTotal: Number(
+                  contactDonationInfo.complete_months_total,
+                ),
+                average: Number(contactDonationInfo.average),
+                minimum: Number(contactDonationInfo.minimum),
+                months: contactDonationInfo.months.map((month, index) => {
+                  const salaryCurrencyTotal = month.donations.reduce(
+                    (convertedTotal, donation) =>
+                      convertedTotal + Number(donation.converted_amount),
+                    0,
+                  );
+                  return {
+                    month: data.months[index],
+                    total: isSalaryType
+                      ? salaryCurrencyTotal
```

Comment:
Why do we have to calculate the month's total manually for salary currency donations? Does the API not do that for us?